### PR TITLE
Build out basic Cypress setup with first E2E tests

### LINKED ISSUE #608: Build out basic Cypress setup with first E2E tests
#### Description

Let's totally revamp our Cypress setup that isn't currently run anywhere and has become totally out-of-date.
We'll still need to hook up a DB solution to run any meaningful tests, but if we can at least take care of a upgrading everything and doing a big cleanup & re-org of what's there now with a basic set of tests that cover the most core flows, that would be a fantastic start to build off of.

### PR DESCRIPTION
**Pull request by robin-macpherson**
## Description

closes #608 

Completely revamps our Cypress set up:
- Major dependency upgrade
- Adds some helpful plugins
- Adds tons of `data-testid`s around the app
- Refactors custom `login` and `signup` commands
- Cleans up all existing test drafts and restructures things so that we just have a single level directory structure within `cypress/integration/` that contains all test files, which are named/organised by user flows
- Writes out basic updates to `.circleci` commented out b/c needs more work to be completed in a separate PR
- Creates skeleton files for future tests to make it as easy as possible to add them as we go

### Does Not
- Finish hooking things up with the new CircleCI job
- Solve DB solution
- Add `data-testid`s to everything

## Subtasks

- [x] I have added this PR to the Journaly Kanban project ✅
- [x] Complete all items listed in description

## Migrations

No migs

### CODE REVIEW

**Review by Lanny — APPROVED** [review]
couple of minor comments, but looks good!

File: cypress/integration/login.spec.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+describe('User logs in', () => {
+  beforeEach(() => {
+    cy.visit('/dashboard/login')
+    cy.setLocalStorage('welcome-modal-july-2020', 'seen')
+  })
+
+  it('Form requires email address', () => {
+    cy.get('[data-testid=login-button]').click()
+    cy.contains('p', 'emailRequiredErrorMessage').should('exist')
```
[the remark this answers is not in the document]
Comment (robin-macpherson) [reply cypress/integration/login.spec.js:9]: Resolving for now per our discussion offline. This is just a temporary thing and we plan to get the translations working properly in the follow-up PR that also addresses other follow-ups

File: cypress/integration/post-feedback.spec.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+describe('Inline Feedback Comments', () => {
+  beforeEach(() => {
+    cy.visit('/dashboard/login')
```
[the remark this answers is not in the document]
Comment (robin-macpherson) [reply cypress/integration/post-feedback.spec.js]: Great call! Removed ✅

[the remark this answers is not in the document]
Comment (robin-macpherson) [reply cypress/integration/post-feedback.spec.js]: Great call! Removed ✅

**Pull request merged**